drivers/SmartThings/zigbee-switch/src/inovelli-vzm31-sn/init.lua

```diff
@@ -0,0 +1,414 @@
+      parameter15 = {parameter_number = 15, size = data_types.Uint8},
+      parameter17 = {parameter_number = 17, size = data_types.Uint8},
+      parameter95 = {parameter_number = 95, size = data_types.Uint8},
+      parameter96 = {parameter_number = 96, size = data_types.Uint8},
+      parameter97 = {parameter_number = 97, size = data_types.Uint8},
+      parameter98 = {parameter_number = 98, size = data_types.Uint8},
+}
+
+local preferences_to_numeric_value = function(new_value)
+  local numeric = tonumber(new_value)
+  if numeric == nil then -- in case the value is Boolean
+    numeric = new_value and 1 or 0
+  end
+  return numeric
+end
+
+local preferences_calculate_parameter = function(new_value, type, number)
+  local numeric = tonumber(new_value)
+  if number == "parameter9" or number == "parameter10" or number == "parameter13" or number == "parameter14"  or number == "parameter15" or number == "parameter55" or number == "parameter56" then
+    if new_value == 101 then
+      return 255
+    else
+      return utils.round(new_value / 100 * 254)
+    end
+  else
+    return new_value
+  end
+end
+
+local is_inovelli_vzm31_sn = function(opts, driver, device)
+  for _, fingerprint in ipairs(INOVELLI_VZM31_SN_FINGERPRINTS) do
+    if device:get_manufacturer() == fingerprint.mfr and device:get_model() == fingerprint.model then
+      local subdriver = require("inovelli-vzm31-sn")
+      return true, subdriver
+    end
+  end
+  return false
+end
+
+local function to_boolean(value)
+  if value == 0 or value =="0" then
+    return false
+  else
+    return true
+  end
+end
+
+local map_key_attribute_to_capability = {
+  [0x00] = capabilities.button.button.pushed,
+  [0x01] = capabilities.button.button.held,
+  [0x02] = capabilities.button.button.down_hold,
+  [0x03] = capabilities.button.button.pushed_2x,
+  [0x04] = capabilities.button.button.pushed_3x,
+  [0x05] = capabilities.button.button.pushed_4x,
+  [0x06] = capabilities.button.button.pushed_5x,
+}
+
+local function button_to_component(buttonId)
+  if buttonId > 0 then
+    return string.format("button%d", buttonId)
+  end
+end
+
+local function scene_handler(driver, device, zb_rx)
+  local bytes = zb_rx.body.zcl_body.body_bytes
+  local button_number = bytes:byte(1)
+  local capability_attribute = map_key_attribute_to_capability[bytes:byte(2)]
+  local additional_fields = {
+    state_change = true
+  }
+
+  local event
+  if capability_attribute ~= nil then
+    event = capability_attribute(additional_fields)
+  end
+
+  local comp = device.profile.components[button_to_component(button_number)]
+  if comp ~= nil then
+    device:emit_component_event(comp, event)
+  end
+end
+
+local function configuration_handler(driver, device, zb_rx)
+  for i,v in ipairs(zb_rx.body.zcl_body.attr_records) do
+    if (v.attr_id.value == 0x000D) then
+    elseif (v.attr_id.value == 0x0015) then
+    else
+    end
+  end
+end
+
+local version_handler = function(driver, device, value, zb_rx)
+end
```

Comment:
nor this one. Is this intentional?

Reply:
Just left over after stripping functionality from the driver. It was left on accident.